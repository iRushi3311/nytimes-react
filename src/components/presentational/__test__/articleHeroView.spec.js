import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from "enzyme";
import {MemoryRouter} from 'react-router-dom';
import ArticleHero from '../article-hero-view.js';
import Article from "../../../models/article";
import AppConstants from '../../../constants/app-constants.js';

import jsdom from 'jsdom';
const doc = jsdom.jsdom('<!doctype html><html><body></body></html>');
const win = doc.defaultView;
global.document = doc;
global.window = win;

describe('>>>C O M P O N E N T S --- snapshot test for ArticleHero', () => {

    let props,
        componentTree;

    beforeEach(() => {
        const src = {
            _id: 12345,
            web_url: 'google.com',
            headline:{
                main: 'So and so guy did bla bla bla'
            },
            snippet: 'body text body text body text body text body text body text ',
            byline:{
                original: 'By RPatel'
            },
            multimedia: [
                {
                    type: 'type-1',
                    subtype: 'subtype-X',
                    url: 'img_X',
                    height: '75px',
                    width: '75px'
                },
                {
                    type: 'type-3',
                    subtype: 'thumbnail',
                    url: 'img_Z',
                    height: '75px',
                    width: '75px'
                },
            ],
            pub_date: '2017-08-17T09:00:01+0000'
        };
        const articleObj = new Article(src);

        props = {
            article: articleObj
        };

        componentTree = renderer.create(
            <MemoryRouter>
                <ArticleHero {...props}/>
            </MemoryRouter>
        ).toJSON();
    });

    it('+++ ArticleHero match with snapshot ', () => {
        expect(componentTree).toMatchSnapshot();
    });

});

describe('>>>C O M P O N E N T S --- mount test for ArticleHero', () => {

    let props,
        mountedArticleHeroView;

    const articleHero = () => {
        if (!mountedArticleHeroView) {
            mountedArticleHeroView = mount(
                <MemoryRouter>
                    <ArticleHero {...props}/>
                </MemoryRouter>
            );
        }
        return mountedArticleHeroView;
    };


    beforeEach(() => {
        const src = {
            _id: 12345,
            web_url: 'google.com',
            headline:{
                main: 'So and so guy did bla bla bla'
            },
            snippet: 'body text body text body text body text body text body text ',
            byline:{
                original: 'By RPatel'
            },
            multimedia: [
                {
                    type: 'type-1',
                    subtype: 'subtype-X',
                    url: 'img_X',
                    height: '75px',
                    width: '75px'
                },
                {
                    type: 'type-3',
                    subtype: 'thumbnail',
                    url: 'img_Z',
                    height: '75px',
                    width: '75px'
                },
            ],
            pub_date: '2017-08-17T09:00:01+0000'
        };
        const articleObj = new Article(src);

        props = {
            article: articleObj
        };

        mountedArticleHeroView = undefined;
    });

    it("always renders a div for ArticleHero", () => {
        const divs = articleHero().find("div");
        const wrappingDiv = divs.first();
        expect(divs.length).toBeGreaterThan(0);
    });

    it("ArticleHero - placeholder thumbnail-img", () => {
        const src = {
            _id: 12345,
            web_url: 'google.com',
            headline:{
                main: 'So and so guy did bla bla bla'
            },
            snippet: 'body text body text body text body text body text body text ',
            byline:{
                original: 'By RPatel'
            },
            multimedia: [],
            pub_date: '2017-08-17T09:00:01+0000'
        };
        const articleObj = new Article(src);

        props = {
            article: articleObj
        };
        const image = articleHero().find(".thumbnail-img");

        expect(image.node.src).toEqual(AppConstants.THUMBNAILPLACEHOLDERURL);
    });

});