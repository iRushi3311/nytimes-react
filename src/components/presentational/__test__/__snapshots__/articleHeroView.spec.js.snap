// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>C O M P O N E N T S --- snapshot test for ArticleHero +++ ArticleHero match with snapshot  1`] = `
<a
  href="/article/12345"
  onClick={[Function]}
>
  <div
    className="hero-contianer article-container"
  >
    <div
      className="hero-title-section"
    >
      <div
        className="media"
      >
        <div
          className="media-body"
        >
          <h3
            className="media-heading title-n1 hero-title-n1"
          >
            So and so guy did bla bla bla
          </h3>
          <h5
            style={
              Object {
                "marginBottom": "0px",
              }
            }
          >
            <span
              className="author-info"
            >
              By RPatel
            </span>
              
            <span
              className="published-time"
            >
              August 17  - 05:00 AM
            </span>
          </h5>
        </div>
        <div
          className="media-right"
        >
          <img
            alt="..."
            className="media-object thumbnail-img"
            src="http://www.nytimes.com/img_Z"
          />
        </div>
      </div>
    </div>
    <div
      className="hero-body-section clip-text"
    >
      body text body text body text body text body text body text 
    </div>
  </div>
</a>
`;
